pkg/kapp/resourcesmisc/kappctrl_k14s_io_v1alpha1_app.go

```diff
@@ -51,12 +51,20 @@ func (s KappctrlK14sIoV1alpha1App) IsDoneApplying() DoneApplyState {
 			return DoneApplyState{Done: false, Message: "Reconciling"}
 
 		case cond.Type == kcv1alpha1.ReconcileFailed && cond.Status == corev1.ConditionTrue:
+			errorMsg := app.Status.UsefulErrorMessage
+			if errorMsg == "" {
+				errorMsg = cond.Message
+			}
```

Comment:
nit: I think we can move this outside of switch 🤔

Reply:
Yep makes sense, that was my initial thought.
I switched up because I thought we would not wanna do this for Reconciling and success scenarios which _should_ be more common. But I am for it in favour of keeping logic deduped